File: FootballApp/src/main/java/com/football/dev/footballapp/dto/TransferDTO.java
package com.football.dev.footballapp.dto;

import com.football.dev.footballapp.models.Club;
import com.football.dev.footballapp.models.Player;
import jakarta.persistence.ManyToOne;

import java.util.Date;

public record TransferDTO(Long id ,
                           Player player,
                           Club previousClub,
                           Club newClub,
                           Date transferDate,
                           Double transferFee
) {
}

File: my-app/src/app/common/transfer-dto.ts
import {PlayerDto} from "./player-dto";
import {ClubDto} from "./club-dto";

export class TransferDto {
   id:number ;
    player:PlayerDto;
   previousClub:ClubDto;
   newClub:ClubDto;
   transferDate:Date;
   transferFee:number;

  constructor(id: number, player:PlayerDto, previousClub: ClubDto, newClub: ClubDto, transferDate: Date, transferFee: number) {
    this.id = id;
    this.player = player;
    this.previousClub = previousClub;
    this.newClub = newClub;
    this.transferDate = transferDate;
    this.transferFee = transferFee;
  }
}
